ggshield/cmd/hmsl/check_secret_manager/hashicorp_vault.py

```diff
@@ -94,5 +123,40 @@ def check_hashicorp_vault_cmd(
             )
 
     vault_token = _get_vault_token(use_cli_token)  # noqa: F841
+    vault_client = VaultAPIClient(url, vault_token)
+
+    # Get mount object and check it exists and we have access to it
+    mount_name, secret_path = _split_vault_mount_and_path(vault_path)
+    all_kv_mounts = list(vault_client.get_kv_mounts())
+    mount = next((item for item in all_kv_mounts if item.name == mount_name), None)
+    if mount is None:
+        raise UnexpectedError(
+            f"mount {mount_name} not found. Make sure it exists and "
+            "that your token has access to it."
+        )
+
+    display_info("Fetching secrets from Vault...")
+    try:
+        result = vault_client.get_secrets(mount, secret_path, recursive)
+    # catch this error here if the path provided is a file that we don't have access to
+    except VaultForbiddenItemError:
+        raise UnexpectedError(
+            "access to the given path was forbidden. "
+            "Are you sure the permissions of the token are correct?"
+        )
+
+    if result.fetching_errors_count > 0:
+        display_warning(
+            f"Could not fetch {result.fetching_errors_count} paths. "
```

Comment:
As discussed, I kept the count display (but as an error instead of a warning) except when verbose is enabled, where I also display the list of paths that could not be scanned.